src/ex1/script.js

```diff
@@ -0,0 +1,100 @@
+const taskInput = document.querySelector(" #taskInput ");
+const addBtn = document.querySelector(" #addBtn ");
+const clearAllBtn = document.querySelector(" #clearAllBtn ");
+const sortAllBtn = document.querySelector(" #sortAllBtn ");
+const taskList = document.querySelector(" #taskList ");
+const placeHolder = document.querySelector(" .placeHolder ")
+const pedingTaskText = document.querySelector(" #peding-task-text ")
+
+const taskArr = JSON.parse(localStorage.getItem("tasks")) || [];
+
+
+//waiting for user input before enable the add button
+taskInput.addEventListener('keyup', () => addBtn.disabled = !taskInput.value)
+
+// add a task when user press Enter key
+taskInput.addEventListener("keypress", (e) => { if (e.key === 'Enter') addNewTask(taskInput.value) });
+
+//add click event listener to add button
+addBtn.addEventListener("click", () => addNewTask(taskInput.value));
+
+//sort the tasks list add them to html list element and save them to the LocalStorage
+sortAllBtn.addEventListener("click", () => {
+    taskArr.sort();
+    taskList.innerHTML = '';
+    insertAllTasks();
+    localStorage.setItem("tasks", JSON.stringify(taskArr)); //add the updated task array to the localStorage
+})
+
+//Clear the tasks list
+clearAllBtn.addEventListener("click", () => {
+    taskArr.length = 0;
+    taskList.innerHTML = ''; //Clear the list of tasks element
+    localStorage.setItem("tasks", JSON.stringify(taskArr)); //add the updated task array to the localStorage
+    updateForm();
+})
+
+//function to add task and update the list and the localStorage
+function addNewTask(taskValue) {
+    taskArr.push(taskValue); //add the user value to the task array
+    taskList.appendChild(createLi(taskValue));
+    taskInput.value = ''; //clear the input after added
+    addBtn.disabled = true;
+    localStorage.setItem("tasks", JSON.stringify(taskArr)); //add the updated task array to the localStorage
+    updateForm();
+}
+
+// function to remove task that have been clicked
+function removeTask(li, taskValue) {
+    taskArr.splice(taskArr.indexOf(taskValue), 1);
+    taskList.removeChild(li);
+    localStorage.setItem("tasks", JSON.stringify(taskArr)); //add the updated task array to the localStorage
+    updateForm();
+}
+
+//update the button and the placeHolder if necessary
+function updateForm() {
```

Comment:
Consider renaming the function to make it more descriptive or even separate into 2 different functions as you always know what you actually need to do beforehand. 1 for hiding the empty state and other for showing it